plugins/woocommerce/includes/class-wc-comments.php

```diff
@@ -215,80 +242,130 @@ public static function clear_transients( $post_id ) {
 		}
 	}
 
+	/**
+	 * Callback for 'wp_insert_comment' to delete the comment count cache if the comment is included in the count.
+	 *
+	 * @param int        $comment_id The comment ID.
+	 * @param WP_Comment $comment    Comment object.
+	 *
+	 * @return void
+	 */
+	public static function increment_comments_count_cache_on_wp_insert_comment( $comment_id, $comment ) {
+		if ( ! self::is_comment_excluded_from_wp_comment_counts( $comment ) ) {
+			$comment_status = wp_get_comment_status( $comment );
+			if ( false !== $comment_status ) {
+				wp_cache_incr( 'wc_count_comments_' . $comment_status, 1, self::COMMENT_COUNT_CACHE_GROUP );
+			}
+		}
+	}
+
+	/**
+	 * Callback for 'comment_status_change' to delete the comment count cache if the comment is included in the count.
+	 *
+	 * @param int|string $new_status The new comment status.
+	 * @param int|string $old_status The old comment status.
+	 * @param WP_Comment $comment    Comment object.
+	 *
+	 * @return void
+	 */
+	public static function update_comments_count_cache_on_comment_status_change( $new_status, $old_status, $comment ) {
+		if ( ! self::is_comment_excluded_from_wp_comment_counts( $comment ) ) {
+			wp_cache_incr( 'wc_count_comments_' . $new_status, 1, self::COMMENT_COUNT_CACHE_GROUP );
+			wp_cache_decr( 'wc_count_comments_' . $old_status, 1, self::COMMENT_COUNT_CACHE_GROUP );
+		}
+	}
+
+	/**
+	 * Determines whether the given comment should be included in the core WP comment counts that are displayed in the
+	 * WordPress admin.
+	 *
+	 * @param WP_Comment $comment Comment object.
+	 *
+	 * @return bool
+	 */
+	private static function is_comment_excluded_from_wp_comment_counts( $comment ) {
+		return in_array( $comment->comment_type, array( 'action_log', 'order_note', 'webhook_delivery' ), true )
+			|| get_post_type( $comment->comment_post_ID ) === 'product';
+	}
+
 	/**
 	 * Delete comments count cache whenever there is
 	 * new comment or the status of a comment changes. Cache
 	 * will be regenerated next time WC_Comments::wp_count_comments()
 	 * is called.
 	 */
 	public static function delete_comments_count_cache() {
-		delete_transient( 'wc_count_comments' );
+		if ( wp_cache_supports( 'flush_group' ) ) {
+			wp_cache_flush_group( self::COMMENT_COUNT_CACHE_GROUP );
```

Comment:
I'm also using this group in https://github.com/woocommerce/woocommerce/pull/56215/files#diff-389d36391ae130391874a998ea6e4a3ab4a6b902f223e6e7c1ba0c464579b03dR18 and I think we'll need to avoid `wp_cache_flush_group( self::COMMENT_COUNT_CACHE_GROUP )`.

Given my other comment prompting `wp_cache_delete_multiple` usage we should be good.